src/odata-to-abstract-sql.ts

```diff
@@ -1034,6 +1043,25 @@ export class OData2AbstractSQL {
 			return this.ReferencedField(prop.resource, prop.name);
 		}
 	}
+
+	Method(prop: any): BooleanTypeNodes | { resource: Resource; name: string } {
+		if (!prop.method) {
+			throw new SyntaxError('Method is missing method entry');
+		}
+
+		if (prop.method[0] != 'call') {
+			throw new SyntaxError(
+				`Invalid value for method invocation: ${prop.method[0]}`,
+			);
+		}
+
+		if (!this.methods[prop.method[1].method]) {
```

Comment:
This should probably be
```suggestion
		if (!this.methods.hasOwnProperty(prop.method[1].method)) {
```
or otherwise it might be possible to do funky things by using a method name that exists on the prototype chain